### PR TITLE
Create basic language generator

### PR DESCRIPTION
**Pull request by xingnanzhou**
- Based on existing implementation, fixed bugs and tested training with GPU
- Verified trained loss has similar results as original code running on CPU
- Original code has get_hidden_states_new method, I'm not sure whether it's actually required. To use this generator with IL or RL, maybe simply get hidden states from forward method is sufficient.

Next steps
- Verify whether we'll need get_hidden_states method when running with IL, RL
- Some part of the code uses open sourced code (copied), check whether we can replace with other implementation
- Decoder code is a little bit complicated, investigate whether we can simply
- Have a test code to evaluate accuracy and/or loss of the model. Currently it takes all training data and no data left for test/validation, so it's hard to know whether the model is overfitting

### CODE REVIEW

**Review by lonelu — APPROVED** [review]

**Review by lonelu — APPROVED** [review]
There is a State encoder in low_level_model. I will remove that one.
